src/internal/m365/onedrive/export.go

```diff
@@ -0,0 +1,160 @@
+package onedrive
+
+import (
+	"context"
+	"strings"
+
+	"github.com/alcionai/clues"
+
+	"github.com/alcionai/corso/src/internal/data"
+	"github.com/alcionai/corso/src/internal/m365/onedrive/metadata"
+	"github.com/alcionai/corso/src/pkg/backup/details"
+	"github.com/alcionai/corso/src/pkg/control"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/path"
+)
+
+// exportCollection is the implementation of data.ExportCollection for OneDrive
+type exportCollection struct {
+	// baseDir contains the path of the collection
+	baseDir string
+
+	// backingCollections will, in most cases contain just one
+	// collection. However, in cases where we have to combine multiple
+	// collections, like when generating pst files for outlook, this
+	// will contain multiple collections.
+	backingCollections []data.RestoreCollection
+
+	// version is the version of the backup this collection was part
+	// of. This is required to figure out how to get the name of the
+	// item.
+	version int
+}
+
+func (ec exportCollection) GetBasePath() string {
+	return ec.baseDir
+}
+
+func (ec exportCollection) GetItems(ctx context.Context) <-chan data.ExportItem {
+	ch := make(chan data.ExportItem)
+
+	go func() {
+		defer close(ch)
+
+		errs := fault.New(false)
+
+		// There will only be a single item in the backingCollections
+		// for OneDrive
+		for _, c := range ec.backingCollections {
+			for item := range c.Items(ctx, errs) {
+				itemUUID := item.UUID()
+				if isMetadataFile(itemUUID, ec.version) {
+					continue
+				}
+
+				name, err := getItemName(ctx, itemUUID, ec.version, c)
+
+				ch <- data.ExportItem{
+					ID: itemUUID,
+					Data: data.ExportItemData{
+						Name: name,
+						Body: item.ToReader(),
```

Comment:
Since this is reading from an unbuffered channel and writing to another unbuffered channel just when it is called, I feel like we don't have to worry about that.

Reply:
Ash has a point here.  `ch` is unbuffered, and we have no guarantees about the consumer keeping up with the channel.  That means we can load many potentially very large items into memory without limit.  A lazy reader is one way to solve this.  Limiting the channel is another.

Reply:
unrelated to whether we use a lazy reader or not, but in our case it's actually the small items that will lead to high memory usage because kopia will load all of the item's data when making the reader. For large items kopia will lazily load data as it goes along

Reply:
IIUC unbuffered channels does not keep any items in the channel and so we will be block and won't create any items unless the user reads from it. That said, I can add in a lazy reader if we should be concerned about user read the data and not using it.

Reply:
oh, doh, you're right.  unbuffered is capacity 1, my bad.